src/sysc_3303_project/performance_tester/PerformanceTester.java

```diff
@@ -0,0 +1,146 @@
+package sysc_3303_project.performance_tester;
+
+import sysc_3303_project.common.events.Event;
+import sysc_3303_project.common.events.EventBuffer;
+import sysc_3303_project.floor_subsystem.FloorEventType;
+
+import java.time.LocalTime;
+import java.util.ArrayList;
+
+/**
+ * @author Robert Simionescu, Ian Holmes
+ *
+ * System for measuring the performance of the system. Measures the time taken to service a request.
+ */
+public class PerformanceTester implements Runnable {
+
+    /**
+     * The filepath for the output file containing the time taken to service a request
+     */
+    public static final String REQUEST_TIME_FILEPATH = "";
+
+    /**
+     * The filepath for the output file containing the time taken to schedule a request
+     */
+    public static final String SCHEDULER_TIME_FILEPATH = "";
+
+    /**
+     * EventBuffer for receiving performance data
+     */
+    private EventBuffer<PerformanceEventType> inputBuffer;
+
+    /**
+     * List of requests that have not finished being serviced.
+     */
+    private ArrayList<PerformanceRequestData> ongoingRequests;
+
+    /**
+     * List of requests that have been read from the input file but not yet scheduled
+     */
+    private ArrayList<PerformanceRequestData> pendingRequests;
+
+    public PerformanceTester(EventBuffer<PerformanceEventType> inputBuffer) {
+        this.inputBuffer = inputBuffer;
+
+    }
+
+    private void outputToFile(String filePath, PerformanceRequestData data, LocalTime servicedTime) {
+
+    }
+
+    /**
+     * Finds the first pending request to the specified destination floor
+     * @param destinationFloor  int, the destination floor of the request
+     * @return  PerformanceRequestData, the first request made to that floor.
+     */
+    private PerformanceRequestData findPendingRequest(int destinationFloor) {
+        for (PerformanceRequestData i : pendingRequests) {
+            if (i.getDestinationFloor() == destinationFloor) {
+                return i;
+            }
+        }
+        return null;
+    }
```

Comment:
Finding the first destination floor that matches seems like a recipe for errors if multiple requests end on the same space. Is this accounted for anywhere or are we just going to not test these cases?

Reply:
this is accounted for in the ongoingRequestServiced() method